docs/u2f_migration.md

```diff
@@ -0,0 +1,157 @@
+
+```ruby
+require 'webauthn'
+require 'webauthn/fake_client'
+require 'webauthn/attestation_statement/fido_u2f'
+
+class U2fCredentialMigrator
+  def initialize(domain:, u2f_certificate:, u2f_key_handle:, u2f_public_key:, u2f_counter:)
+    @domain = domain
+    @u2f_certificate = u2f_certificate
+    @u2f_key_handle = u2f_key_handle
+    @u2f_public_key = u2f_public_key
+    @u2f_counter = u2f_counter
+  end
+
+  def authenticator_data
+    @authenticator_data ||= WebAuthn::FakeAuthenticator::AuthenticatorData.new(
+      rp_id_hash: OpenSSL::Digest::SHA256.digest(@domain.host),
+      credential: {
+        id: credential_id,
+        public_key: credential_cose_key
+      },
+      sign_count: @u2f_count,
+      user_present: true,
+      user_verified: false,
+      aaguid: WebAuthn::AttestationStatement::FidoU2f::VALID_ATTESTED_AAGUID,
+    )
+  end
+
+  def credential
+    @credential ||= begin
+      hash = authenticator_data.send(:credential)
+      WebAuthn::AuthenticatorData::AttestedCredentialData::Credential.new(hash[:id], hash[:public_key].serialize)
+    end
+  end
+
+  def attestation_type
+    WebAuthn::AttestationStatement::ATTESTATION_TYPE_BASIC_OR_ATTCA
+  end
+
+  def attestation_trust_path
+    @attestation_trust_path ||= [OpenSSL::X509::Certificate.new(Base64.strict_decode64(@u2f_certificate))]
+  end
+
+  private
+
+  # Let credentialId be a credentialIdLength byte array initialized with CTAP1/U2F response key handle bytes.
+  def credential_id
+    Base64.urlsafe_decode64(@u2f_key_handle)
+  end
+
+  # Let x9encodedUserPublicKey be the user public key returned in the U2F registration response message [U2FRawMsgs].
+  # Let coseEncodedCredentialPublicKey be the result of converting x9encodedUserPublicKey’s value from ANS X9.62 /
+  # Sec-1 v2 uncompressed curve point representation [SEC1V2] to COSE_Key representation ([RFC8152] Section 7).
+  def credential_cose_key
+    decoded_public_key = Base64.strict_decode64(@u2f_public_key)
+    if WebAuthn::AttestationStatement::FidoU2f::PublicKey.uncompressed_point?(decoded_public_key)
+      COSE::Key::EC2.new(
+        alg: COSE::Algorithm.by_name("ES256").id,
+        crv: 1,
+        x: decoded_public_key[1..32],
+        y: decoded_public_key[33..-1]
+      )
+    else
+      raise "expected U2F public key to be in uncompressed point format"
+    end
+  end
+end
+```
+
+Of course a `verify` instance method is not implemented, as there is no real interaction with an interaction going on.
+Like the readme prescribes, you need to store at least the object's `credential.id` and `credential.public_key` in your
+database. 
+
+For reasons explained further below, it is also recommended to store a boolean value whether or not a stored credential
+if migrated from U2F or not. For this documented, we'll assume this database column is called `u2f_migrated`.
+
+```ruby
+migrated_credential_response = U2fCredentialMigrator.new(
```

Comment:
What would you think of providing a helper function to construct `U2fCredentialMigrator` from the `domain` and `u2f_registration` rather than requiring the caller to deconstruct the `u2f_registration` manually?